map-storage/src/Services/CustomEntityCollectionService.ts

```diff
@@ -0,0 +1,113 @@
+import {
+    CollisionGrid,
+    ENTITIES_FOLDER_PATH,
+    ENTITY_COLLECTION_FILE,
+    EntityCollectionRaw,
+    EntityRawPrefab,
+    mapCustomEntityDirectionToDirection,
+} from "@workadventure/map-editor";
+import { DeleteCustomEntityMessage, ModifyCustomEntityMessage, UploadEntityMessage } from "@workadventure/messages";
+import { fileSystem } from "../fileSystem";
+import { mapPathUsingDomainWithPrefix } from "./PathMapper";
+
+export class CustomEntityCollectionService {
+    private readonly hostname: string;
+
+    constructor(hostname: string) {
+        this.hostname = hostname;
+    }
+
+    private getEntityCollectionFileVirtualPath() {
+        return mapPathUsingDomainWithPrefix(`${ENTITIES_FOLDER_PATH}/${ENTITY_COLLECTION_FILE}`, this.hostname);
+    }
+
+    private getEntityToUploadVirtualPath(fileName: string) {
+        return mapPathUsingDomainWithPrefix(`${ENTITIES_FOLDER_PATH}/${fileName}`, this.hostname);
+    }
+
+    public async uploadEntity(uploadEntityMessage: UploadEntityMessage) {
+        const { imagePath, file } = uploadEntityMessage;
```

Comment:
The imagePath comes from the front.
It could be weaponized against us.

For instance, a hacker could pass something like "../../../etc/passwd" to overwrite the user list, etc....
I think we should apply some security checks here:

- The file must not contain ".." or "/" in the name
- The file must not already exist on the file system (no overwriting files, uploads are necessarily new files)
- The extension should be a valid image extension.